skietbaan_fe/src/components/AddGroup.js

```diff
@@ -19,52 +24,85 @@ class AddGroup extends Component {
     this.setState({ [e.target.name]: e.target.value });
   }
   onClick() {
-    let RequestObject = {
-      Name: this.state.name
-    };
-    this.props.groupActions.getname(RequestObject);
-    history.push("/Groups");
+    if (this.state.groups.indexOf(this.state.name) == -1) {
+    
+      if (this.state.name.length != 0) {
+        this.props.getName(this.state.name);
+        history.push("/Groups");
+      
+      } else {
+        this.setState({ txt: "group name can't be empty" });
+      }
+    } else {
+      this.setState({ exist: false });
+    }
+  }
+
+  UNSAFE_componentWillMount() {
```

Comment:
PLease change this to `unsafeComponentWillMount`

Reply:
it doesn't work if i take it put